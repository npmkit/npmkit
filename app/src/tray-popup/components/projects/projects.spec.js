import { mount } from 'enzyme';
import { Provider } from 'unstated';

const mockedElectron = {
  remote: {
    app: {
      getPath() {
        return '/fake/path';
      },
    },
  },
  ipcRenderer: {
    on() {},
    send() {},
  },
};

const mockedStore = class MockedStore {
  set() {}
  get() {}
};

describe('<Projects />', () => {
  beforeAll(() => {
    jest.mock('electron', () => mockedElectron);
    jest.mock('electron-store', () => mockedStore);
  });

  afterAll(() => {
    jest.unmock('electron');
    jest.unmock('electron-store');
  });

  it('matches snapshot', () => {
    const AppContainer = require('~/common/app-container').default;
    const Projects = require('./').default;
    expect(
      mount(
        <Provider inject={[new AppContainer()]}>
          <Projects />
        </Provider>
      ).html()
    ).toMatchSnapshot();
  });
});
