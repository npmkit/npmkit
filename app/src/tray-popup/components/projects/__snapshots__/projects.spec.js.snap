// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> matches snapshot 1`] = `"<div data-test-id=\\"projects\\" class=\\"sc-fjdhpX bQFtOV\\"><div class=\\"sc-iwsKbI jvxxtM\\"><div class=\\"sc-gZMcBi mmRYP\\"></div><div class=\\"sc-gqjmRU iZaeOp\\"><div class=\\"sc-VigVT dPWYfU\\"></div><div class=\\"sc-jTzLTM kKfebK\\"></div></div></div><div class=\\"sc-iwsKbI jvxxtM\\"><div class=\\"sc-gZMcBi mmRYP\\"></div><div class=\\"sc-gqjmRU iZaeOp\\"><div class=\\"sc-VigVT dPWYfU\\"></div><div class=\\"sc-jTzLTM kKfebK\\"></div></div></div><div class=\\"sc-iwsKbI jvxxtM\\"><div class=\\"sc-gZMcBi mmRYP\\"></div><div class=\\"sc-gqjmRU iZaeOp\\"><div class=\\"sc-VigVT dPWYfU\\"></div><div class=\\"sc-jTzLTM kKfebK\\"></div></div></div></div>"`;
